### ISSUE TITLE
Front End and Design process

### ISSUE THREAD

**Issue by rohit121** [storyscript/community#11]
tl:dr;  We are now transitioning to a better process for front end development. 

### Notes

Couple of things highlighted that are a pain point right now - 
Quality of designs being handed over to engineers - we need to spend more time in the ideation process, mock the entire interaction end to end to make sure we get the finer details right such as popups and alert messages etc. Also this would make the flows easy to understand as well for everyone involved. 

Quality of code being pushed out and testing process. We should have caught a lot of small things, bugs that made it production. We can fix this by adding some test cases and process on the front end.

The other issue is we need to react much faster to production bugs and get into the habit of deploying more and iterating faster. 

### Result
Spend more time at the ideation phase, mock all interactions, popups, error messages etc.
No development work to begin before there is a sign off from all stakeholders.  

@JeanBarriere @Arinono  to evaluate percy and put in some ground work this sprint.

Invision to be the place where all designs and interactions will be shared for comments, reviews and feedback will be given. Figma is the last step and to be only used by engineers when they are coding.

Also spend sometime to get the polish @stevepeak is looking for - We will have a meeting this week where steve can walk us through on how to develop a better eye for the finer things.

----

> This meeting took place in Office with the following attendeers: @stevepeak @JeanBarriere @Arinono @rohit121 
FYI @aydaoz @matthewhudson

**Comment by aydaoz** [comment]
Agree. we always have to create a guideline (popups and alert messages etc.) as I did for OMG app. but didn't had time for SS dash at the beginning. Tbh, if nothing rushed and given enough time, ex. more flexible deadlines for big projects, I believe all details will be caught by us.

**Comment by JeanBarriere** [comment]
> Invision to be the place where all designs and interactions will be shared for comments, reviews and feedback will be given. Figma is the last step and to be only used by engineers when they are coding.

I honestly don't see a change here. it's what it should be today, isn't it the case ?

**Comment by Arinono** [comment]
>@JeanBarriere @Arinono to evaluate percy and put in some ground work this sprint.

E2E and Percy support are now added to the [dashboard](https://github.com/storyscript/dashboard/pull/150) iteration process.

Now the longer work on this topic is to add more E2E tests. And for future feature to define them in a way where we can write those E2E tests before starting to code the feature

**Comment by JeanBarriere** [comment]
> > @JeanBarriere @Arinono to evaluate percy and put in some ground work this sprint.
> 
> E2E and Percy support are now added to the [dashboard](https://github.com/storyscript/dashboard/pull/150) iteration process.
> 
> Now the longer work on this topic is to add more E2E tests. And for future features to define them in a way where we can write those E2E tests before starting to code the feature

Amazing work here. The story-based issue definition we're experiencing on the dashboard can be a way to make this easier.

**Comment by stevepeak** [comment]
A new design process (more generally project management process) is being worked on. 👍
